src/Common/src/TypeSystem/IL/Stubs/ILEmitter.cs

```diff
@@ -201,6 +204,189 @@ public void EmitSwitch(ILCodeLabel[] labels)
             }
         }
 
+        public void EmitLdInd(TypeDesc type)
+        {
+            switch (type.Category)
```

Comment:
Can this switch on `type.Category.UnderlyingType`? It would force enums to use the more efficient encoding.